### PR TITLE
Simplify PR template, add checkboxes

### PR DESCRIPTION
**Pull request by izgeri**
Simplify the PR template to reduce the number of sections users are asked to fill out
Add checkboxes for the handful of primary concerns we'd like users to verify when
submitting a PR

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
-> Does the knowledge base need an update?
+#### Change log
+- [ ] The CHANGELOG has been updated, or
+- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update
+
+#### Test coverage
+- [ ] This PR includes new unit or integration tests to go with the code changes
+
+#### Follow-on issues
+- [ ] Follow-up issue(s) have been logged to update documentation or related projects, or
```
Comment (jvanderhoof) [inline .github/PULL_REQUEST_TEMPLATE.md]: What do you think about having the PR include links to followup tickets? I wonder if it'll help insure these issues are actually filed.
```suggestion
- [ ] Follow-up issue(s) have been logged (and links included below) to update documentation or related projects, or
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
-> Has this change been documented (Readme, docs, etc.)?
+### Checklists
 
-> Does the knowledge base need an update?
+#### Change log
+- [ ] The CHANGELOG has been updated, or
+- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update
+
+#### Test coverage
+- [ ] This PR includes new unit or integration tests to go with the code changes
```
Comment (jvanderhoof) [inline .github/PULL_REQUEST_TEMPLATE.md]: I have a strong preference towards Unit tests in most cases.  It feels like our integration test first approach has hindered our ability to get fast tests locally.  What do you think about changing `or` to `and`?
```suggestion
- [ ] This PR includes new unit and integration tests to go with the code changes
```

**Comment by izgeri** [comment]
@jvanderhoof I accepted your revisions - can you please re-review? if the build is green by the time you take a look, please squash merge. if not, I can merge once it is green. thanks!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
-> Has this change been documented (Readme, docs, etc.)?
+### Checklists
 
-> Does the knowledge base need an update?
+#### Change log
+- [ ] The CHANGELOG has been updated, or
+- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update
+
+#### Test coverage
+- [ ] This PR includes new unit and integration tests to go with the code changes
```
Comment (micahlee) [inline .github/PULL_REQUEST_TEMPLATE.md]: Would it be worth adding the negative option to check here as well? E.g.
```
 - [ ] No new or updated tests are required
```

This  would help distinguish between forgotten tests versus the author doesn't believe any new tests are are needed.

Comment (izgeri) [reply .github/PULL_REQUEST_TEMPLATE.md]: I can add it, and considered it. I suppose that in leaving it without the negative I intended to imply that the negative case was not a thing, but of course it sometimes is (not least for my own work in this repo, which doesn't usually involve changing testable code).

I'll add it.

Comment (izgeri) [reply .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests
```

Comment (izgeri) [reply .github/PULL_REQUEST_TEMPLATE.md]: @micahlee let me know how that looks

Comment (micahlee) [reply .github/PULL_REQUEST_TEMPLATE.md]: Looks good to me, thanks!

**Comment by izgeri** [comment]
@micahlee can I get a new approval, please? I made the change to the test checkboxes. thanks!!

**Pull request merged**